_posts/2019-03-06-generating-print-quality-plots.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: Generating print-quality plots in R
+author: Danny Eisenberg
+categories: plots, R, RStudio
+---
+Some students have mentioned that they had generated plots with R and submitted them in their theses, but were requested to redo the plots at high resolution and to change the colour palette.
+
+## High Resolution
+
+If you use RStudio, you can export your plots to a file in either PDF or PNG format. If you save it in PDF format, by default it is high resoultion. If, however, you wish to save it as a PNG bitmap, the default resolution will be 72 dpi (dots per inch), the standard screen resolution, far too low for high quality printing, which needs to be at least 300 dpi.
+
+To see what I mean, let's generate a plot as an example. We are will use a sample file recording maximum temperature available from the [UK Met Office](https://www.metoffice.gov.uk/hadobs/hadghcnd/download.html): https://www.metoffice.gov.uk/hadobs/hadghcnd/data/HadGHCND_1949-2011_TXx.nc.gz
+
+We will now plot make a contour map displaying the average maximum temperature for the month of January:
+```R
+library(ncdf4)
+library(maps)
+
+# load the heatwave data file
+ncname <- "HadGHCND_1949-2011_TXx_1.nc"
+ncin <- nc_open(ncname)
+
+# extract lat, lon and January Tmax variables
+lats <- rev(ncvar_get(ncin, "lat"))
+lons <- ncvar_get(ncin, "lon")
+janVals <- ncvar_get(ncin, "January")
+
+# set up the matrix for the contour map
+area <- matrix(0,nrow=dim(lons), ncol=dim(lats))
+for (lat in 1:dim(lats)){
+  for (lon in 1:dim(lons)){
+    vals <- janVals[lon, lat,]
+    vals <- vals[!is.na(vals)]
+    area[lon, lat] <- mean(vals)
+  }
+}
+
+filled.contour(lons,lats,area,
+                   xlim = range(lons), ylim = range(lats),
+                   plot.axes = {axis(1);axis(2);
+                     map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+```
```

Comment:
For example, I can get make a plot (not sure if it is suitable for your purposes) like so:
```R
ncin <- nc_open('http://dapds00.nci.org.au/thredds/dodsC/ua8/ARCCSS_Data-1/v2-0/001GPsc_E0.monthly_TS.1978_2011.nc')
lats <- ncvar_get(ncin, "lat")
lons <- ncvar_get(ncin, "lon")
tas <- ncvar_get(ncin, "tas", count=c(192, 145, 1))
filled.contour(lons, lats, tas, plot.axes = {axis(1);axis(2);
                     map('world',add=TRUE, wrap=c(0,360), interior = FALSE)})
```